src/pages/fundamentals/eutxo-patterns.mdx

```diff
@@ -29,23 +29,231 @@ the transaction contain the corresponding `Input`. By doing this, the minting po
+
+      todo @"Check if output is consumed" 
+
+    }
+  }
+}
+```
+To enforce uniqueness, we need to ensure that the UTxO defined as OutputReference in the validator parameters is 
+consumed. This is because every OutputReference is a unique combination of the Transaction ID and an Output Index
+Integer. It's important to remember that the Transaction ID is a `Hash<Blake2b_256, Transaction>`, which is also a
+unique identifier and will not be repeated.
+
+``` aiken
+validator(utxo_ref: OutputReference, token_name: AssetName) {
+  fn nft(action: Action, ctx: ScriptContext) -> Bool {
+      expect Mint(policy_id) = ctx.purpose
+
+      let ScriptContext { transaction: Transaction { inputs, mint, .. }, .. } = ctx
+
+      expect [(asset_name, amount)] = mint |> from_minted_value |> tokens(policy_id) |> to_list() 
+
+      // Check if the specified UTxO reference (utxo_ref) is consumed by any input
+      let is_output_consumed = list.any(inputs, fn(input) { input.output_reference == utxo_ref })
+
+      when action is {
+        Minting ->
+          is_output_consumed? && (1 == amount)? && (asset_name == token_name)?
+
+        Burning -> (-1 == amount)? && (asset_name == token_name)? // No need to check if output is consumed for burning
+      }
+    }
+  }
+}
+```
+
 ### Receipts
 
-A validator can mint a unique receipt for a transaction, by requiring that the name of the asset is any
+A validator can mint a unique receipt for a transaction by requiring that the name of the asset is any
 unique value specific to the transaction where validation is set to occur. In other words, if we enforce
 that only one receipt is to be minted per transaction, we can use `blake2b_256` and `cbor.serialise` to
-get a unique value which can be assigned to the `AssetName` expected for the receipt from the
+get a unique value that can be assigned to the `AssetName` expected for the receipt from the
 `OutputReference` from the first `Input` in our transactions inputs.
 
+``` aiken
+validator {
+  fn receipt(_r: Data, ctx: ScriptContext) -> Bool {
+    // This validator expects a minting transaction
+    expect Mint(policy_id) = ctx.purpose
+    let ScriptContext { transaction: Transaction { inputs, mint, .. }, .. } =
+      ctx
+
+    // Select the first input and concatenate its output reference's index and tx_id hash, converting it to a byte array
+    expect [Pair(asset_name, amount)] =
+      mint |> from_minted_value |> tokens(policy_id) |> to_pairs()
+    expect Some(first_input) = list.at(inputs, 0)
+    let first_input_hash = first_input.output_reference.transaction_id.hash
+    let first_input_index =
+      to_bytearray(from_int(first_input.output_reference.output_index))
+    let first_input_out_ref = concat(first_input_hash, first_input_index)
+
+    // Compare the asset name with the output reference as a byte array
+    asset_name == first_input_out_ref && amount == 1
```

Comment:
This won't work because of the limitation in asset name sizes.

https://gist.github.com/Riley-Kilgore/9651d6489b5b47cf048cfdbb1c4531d5

This gist demonstrates what we have to do for the receipt pattern.